pyscf/tools/trexio.py

```diff
@@ -27,15 +29,39 @@ def to_trexio(obj, filename, backend='h5'):
             raise NotImplementedError(f'Conversion function for {obj.__class__}')
 
 def _mol_to_trexio(mol, trexio_file):
-    trexio.write_nucleus_num(trexio_file, mol.natm)
+    # 1 Metadata
+    trexio.write_metadata_code_num(trexio_file, 1)
+    trexio.write_metadata_code(trexio_file, [f'PySCF-v{pyscf.__version__}'])
+    #trexio.write_metadata_package_version(trexio_file, f'TREXIO-v{trexio.__version__}')
 
+    # 2 System
+    trexio.write_nucleus_num(trexio_file, mol.natm)
+    nucleus_charge = [mol.atom_charge(i) for i in range(mol.natm)]
+    trexio.write_nucleus_charge(trexio_file, nucleus_charge)
+    trexio.write_nucleus_coord(trexio_file, mol.atom_coords())
     labels = [mol.atom_pure_symbol(i) for i in range(mol.natm)]
     trexio.write_nucleus_label(trexio_file, labels)
-    if mol._ecp:
-        raise NotImplementedError
-    trexio.write_nucleus_charge(trexio_file, mol.atom_charges())
-    trexio.write_nucleus_coord(trexio_file, mol.atom_coords())
+    if mol.symmetry:
+        trexio.write_nucleus_point_group(trexio_file, mol.groupname)
 
+    # 2.3 Periodic boundary calculations
+    try:
```

Comment:
If you don't need to consider the compatibility of old pyscf versions, PBC can be checked using isinstance(mol, pbc.gto.Cell) . Starting from pyscf-2.5, gto.Mole and pbc.gto.Cell represent two distinct classes.

Reply:
Dear @sunqm, Sure, Thank you very much. I am now implementing the PBC part, in which I do this check as you proposed.